tests/lib/rules/no-octal-escape.js

```diff
@@ -24,13 +24,31 @@ ruleTester.run("no-octal-escape", rule, {
         "var foo = \"\\x51\";",
         "var foo = \"foo \\\\251 bar\";",
         "var foo = /([abc]) \\1/g;",
-        "var foo = '\\0';"
+        "var foo = '\\0';",
+        "'\\0 '",
+        "'\\0a'",
+        "'\\\\1'",
+        "'\\\\01'",
+        "'\\08'",
+        "'\\09'"
     ],
     invalid: [
         { code: "var foo = \"foo \\01 bar\";", errors: [{ message: "Don't use octal: '\\01'. Use '\\u....' instead.", type: "Literal" }] },
         { code: "var foo = \"foo \\251 bar\";", errors: [{ message: "Don't use octal: '\\251'. Use '\\u....' instead.", type: "Literal" }] },
         { code: "var foo = \"\\751\";", errors: [{ message: "Don't use octal: '\\75'. Use '\\u....' instead.", type: "Literal" }] },
         { code: "var foo = \"\\3s51\";", errors: [{ message: "Don't use octal: '\\3'. Use '\\u....' instead.", type: "Literal" }] },
-        { code: "var foo = \"\\\\\\751\";", errors: [{ message: "Don't use octal: '\\75'. Use '\\u....' instead.", type: "Literal" }] }
+        { code: "var foo = \"\\\\\\751\";", errors: [{ message: "Don't use octal: '\\75'. Use '\\u....' instead.", type: "Literal" }] },
+        { code: "'\\0\\1'", errors: [{ message: "Don't use octal: '\\1'. Use '\\u....' instead.", type: "Literal" }] },
```

Comment:
Just a minor style note, but since you are using messageId now, may as well take advantage of the messageId inside of your tests (makes the test file a little more DRY) :bowing_man:

```
{ code: "'\\0\\1'", errors: [{ messageId: "octalEscapeSequence", data: {sequence: "\\1"}] }
```

Also, is specifying the type necessary? I think the rule tester runs rules in isolation of any other rule and you're guaranteed that the node type is a literal given that it's the only context provided inside the return block :smile: (although admittedly, if it's running on Identifiers or other types we would be in trouble!)

Reply:
> Just a minor style note, but since you are using messageId now, may as well take advantage of the messageId inside of your tests (makes the test file a little more DRY) 
> 
> ```
> { code: "'\\0\\1'", errors: [{ messageId: "octalEscapeSequence", data: {sequence: "\\1"}] }
> ```

I agree, but before changing all tests (including the already existing ones) I must first ask for confirmation that id + data would be preferable over full messages (perhaps one or two should stay to test the message itself).

> Also, is specifying the type necessary? I think the rule tester runs rules in isolation of any other rule and you're guaranteed that the ASTNode type is a Literal given that it's the only context provided inside the return block  (although admittedly, if it's running on Identifiers or other types we would be in trouble!)

I believe this should stay, to be implementation independent and also cover future modifications, no matter how obvious it is with the current version :)